reactivesocket-client/src/main/java/io/reactivesocket/client/LoadBalancer.java

```diff
@@ -121,18 +174,43 @@ public LoadBalancer(Publisher<List<ReactiveSocketFactory<SocketAddress>>> factor
     }
 
     private synchronized void addSockets(int numberOfNewSocket) {
-        activeFactories.entrySet()
-            .stream()
-            // available factories that don't map to an already established socket
-            .filter(e -> !activeSockets.containsKey(e.getKey()))
-            .map(e -> e.getValue())
-            .filter(factory -> factory.availability() > 0.0)
-            .sorted((a, b) -> -Double.compare(a.availability(), b.availability()))
-            .limit(numberOfNewSocket)
-            .forEach(factory -> {
-                pendingSockets += 1;
-                factory.apply().subscribe(new SocketAdder(factory.remote()));
-            });
+        int n = Math.min(activeFactories.size(), numberOfNewSocket);
+        int maxEffort = 1 + activeFactories.size()/2;
+        int effort = 0;
+        while (n > 0 && effort < maxEffort) {
```

Comment:
In practice, I don't think it's necessary.
If `aperture` is bigger than the number of hosts, it will just mean that we'll have (on average) more than 2 outstanding messages.

The only case where this could be problematic is when you have a very small number of clients/servers, and because of how Netty assign connections to Threads, if a 32 cores server only has 2 connections, it will only use 1/16th of its CPU.